lib/kernel/src/file.erl

```diff
@@ -1413,7 +1413,7 @@ path_open_first([Path|Rest], Name, Mode, LastError) ->
 	    case open(FileName, Mode) of
 		{ok, Fd} ->
 		    {ok, Fd, FileName};
-		{error, enoent} ->
+		{error, E} when E == enoent; E == enotdir ->
```

Comment:
Maybe we should use `Reason` (`R`) instead of `E` thus it'll be closer to what spec says—`{error, Reason}`—for `open/2`? :bowtie: